stream-chat-android-client/src/main/java/io/getstream/chat/android/client/call/CoroutineCall.kt

```diff
@@ -0,0 +1,36 @@
+package io.getstream.chat.android.client.call
+
+import io.getstream.chat.android.client.internal.InternalStreamChatApi
+import io.getstream.chat.android.client.utils.Result
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.withContext
+
+@InternalStreamChatApi
+public class CoroutineCall<T : Any>(
+    private val scope: CoroutineScope,
+    private val runnable: suspend () -> Result<T>
```

Comment:
This `runnable` has to support proper coroutine cancellation for cancellation to work nicely, this is up to whoever creates instances of `CoroutineCall`